internal/pkg/template/workload.go

```diff
@@ -456,6 +491,17 @@ type NetworkLoadBalancer struct {
 	CertificateRequired bool
 }
 
+// ApplicationLoadBalancer holds configuration that's needed for an Application Load Balancer.
+type ApplicationLoadBalancer struct {
```

Comment:
My impression from reading the templates that this corresponds more to:
```go
type ALBListener struct // or alternatively "HTTPListener"
```

The struct contains fields that are relevant to all the listener rules under a listener. ALB level config such as subnets and security groups are not relevant going to be in this struct.